### ISSUE TITLE
Lookup fails if looked up field name contains a dot

### ISSUE THREAD

**Issue by jwoLondon** [vega/vega-lite#7867]
The following fails to render because the lookup field (`y.y`) contains a dot  ([editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoGsrQBGJQA8LABgC+Ck2dQAma2lcOnIU2gDMHqj+DgC6jiCYAE5IshAAZmxRCEaggmxsANYMOGi+ZkrxUWwpmipqGlo6egaoxvkWSgCelE0B3s6eza2eHQ3B3W2o5mERmXBDDYU0cILKtYYgLW3h9qFKcLJQbMo0smRooDZl8TNzeTZKmE04cHkAjgyxmHRqNKQgEUOgp7MuS4pIjc7uhHs9Xi8PvYIsgopk8lAaFEoII7vYgA)).

It looks like the field name isn't being escaped as it generates the error `undefined is not an object (evaluating '_["y"]["y"]')`

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "data": {
    "values": [
      {"id": 1, "x": 10},
      {"id": 2, "x": 20},
      {"id": 3, "x": 30}
    ]
  },
  "transform": [
    {
      "lookup": "id",
      "from": {
        "data": {
          "values": [
            {"id": 1, "y.y": 30},
            {"id": 2, "y.y": 20},
            {"id": 3, "y.y": 10}
          ]
        },
        "key": "id",
        "fields": ["y.y"]
      }
    }
  ],
  "encoding": {
    "x": {"field": "x","type": "quantitative"},
    "y": {"field": "y","type": "quantitative"}
  },
  "mark": "circle"
}
```

**Comment by domoritz** [comment]
Since You don't want a nested access, I think you need to escape the `.` as in `"fields": ["y.y"]` -> `"fields": ["y\\.y"]`. [Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoGsrQBGJQA8LABgC+Ck2dQAma2lcOnIU2gDMHqj+DgC6jiCYAE5IshAAZmxRCEaggmxsANYMOGi+ZkrxUWwpmipqGlo6egaoxvkWSgCelE0B3s6eza2eHQ3B3W2o5mERmXBDDYU0cILKtYYgTQA6yz3h9qFKcLJQbMo0smRooDZl8TNzeTZKmE04cHkAjgyxmHRqNKQgEUOgF7MXEserd7o90C83h93t97BFkFFMnkoDQolBBI97EA).

I'm not sure whether Vega-Lite should do this automatically. Maybe it should not since you may read a nested field in a lookup. The behavior may be expected.

**Comment by jwoLondon** [comment]
You are right, when I tried this I forgot to update the escaped field name in the position encoding. And I agree that we may want to reference nested object so conversion should not be automatic.

However it may be useful to catch the fatal 'undefined is not an object' error and treat it instead as any other reference to a non-existent field.

**Comment by domoritz** [comment]
Agreed. That's an issue with Vega. I will move the issue.
